tests/ca_test.go

```diff
@@ -215,4 +215,70 @@ func Test_CA(t *testing.T) {
 
 		ctx.Req.Equal(expectedName, identity.Path("data.name").Data().(string))
 	})
+
+	t.Run("CAs with auth enabled can authenticate", func(t *testing.T) {
+		ctx.testContextChanged(t)
+		ca := newTestCa()
+
+		ca.id = ctx.AdminSession.requireCreateEntity(ca)
+
+		caValues := ctx.AdminSession.requireQuery("cas/" + ca.id)
+		verificationToken := caValues.Path("data.verificationToken").Data().(string)
+		ctx.Req.NotEmpty(verificationToken)
+
+		validationAuth := ca.CreateSignedCert(verificationToken)
+		clientAuthenticator := ca.CreateSignedCert(eid.New())
+
+		resp, err := ctx.AdminSession.newAuthenticatedRequest().
+			SetHeader("content-type", "text/plain").
+			SetBody(validationAuth.certPem).
+			Post("cas/" + ca.id + "/verify")
+
+		ctx.Req.NoError(err)
+		ctx.logJson(resp.Body())
+		ctx.Req.Equal(http.StatusOK, resp.StatusCode())
+
+		ctx.completeCaAutoEnrollment(clientAuthenticator)
+
+		enrolledSession, err := clientAuthenticator.Authenticate(ctx)
+
+		ctx.Req.NoError(err)
+		ctx.Req.NotEmpty(enrolledSession)
+
+		t.Run("CAs with auth disabled can no longer authenticate", func(t *testing.T) {
+			ctx.testContextChanged(t)
+			ca.isAuthEnabled = false
+			resp := ctx.AdminSession.patchEntity(ca, "isAuthEnabled")
+			ctx.Req.NotEmpty(resp)
+			ctx.Req.Equal(http.StatusOK, resp.StatusCode())
+
+			enrolledSession, err := clientAuthenticator.Authenticate(ctx)
+
+			ctx.Req.Error(err)
+			ctx.Req.Empty(enrolledSession)
+		})
+
+		t.Run("CAs with auth re-enabled an authenticate", func(t *testing.T) {
+			ctx.testContextChanged(t)
+			ca.isAuthEnabled = true
+			resp := ctx.AdminSession.patchEntity(ca, "isAuthEnabled")
+			ctx.Req.NotEmpty(resp)
+			ctx.Req.Equal(http.StatusOK, resp.StatusCode())
+
+			enrolledSession, err := clientAuthenticator.Authenticate(ctx)
+
+			ctx.Req.NoError(err)
+			ctx.Req.NotEmpty(enrolledSession)
+		})
+
+		t.Run("deleting a CA no longer allows authentication", func(t *testing.T) {
+			ctx.testContextChanged(t)
+			ctx.AdminSession.requireDeleteEntity(ca)
+
+			enrolledSession, err := clientAuthenticator.Authenticate(ctx)
+
+			ctx.Req.Error(err)
+			ctx.Req.Empty(enrolledSession)
+		})
```

Comment:
does it make sense to add a test here for "client has a session, remove the CA, session is removed/disconnected"? or is that somehow covered in these tests already?

Reply:
I looked through this and I'm adding another ticket to address this.

The system doesn't do that. Pretty sure it never did. Also, there isn't a simple lookup to determine which API sessions to delete, but it can be added. I also was thinking if we disable auth on a CA does that also terminate existing API sessions and sessions?

Reply:
seems like it should to me. might be a footgun but if it's easy enough to do then I think it makes sense

Reply:
https://github.com/openziti/edge/issues/311